### PR TITLE
Implement analyze, discard, explain, execute, prepare, prepare transaction

### PR DESCRIPTION
**Pull request by mebelousov**
1. Elapsed time grew up abnormally.
Mainly SelectCommandTest grew up from 11.192 sec  to 227.417 sec.
I have no idea about the reasons. I tested some commits related to `expr`. There was growth only on 10 second.
Need in-depth exploration.
2. I only moved files to folder "prepare_transaction". I didn't run `scripts/gen_psql_tests`

### CODE REVIEW

File: antlr4/PostgreSQLLexer.g4 [hunk 1]
```diff
@@ -343,6 +347,7 @@ INTERVAL                             : I N T E R V A L;
 INTO                                 : I N T O;
 INVOKER                              : I N V O K E R;
 IS                                   : I S;
+ISNOTNULL                            : I S N O T N U L L;
```
Comment (tshprecher) [inline antlr4/PostgreSQLLexer.g4]: is this really a standalone symbol? Can you make sure you're checking against version 10.6 of the postgres documentation? I know that this is a little behind the latest, but I want to make sure this is done well before tweaking the grammar for later versions. I'm running the v10.6 of postgres on my local host, and this is what I see when I use `ISNOTNULL` vs `IS NOT NULL`:

```
postgres=# select 1 where 1 isnotnull;                                                                                                                          ERROR:  syntax error at or near "isnotnull"
LINE 1: select 1 where 1 isnotnull;
                         ^
postgres=# select 1 where 1 is not null;                                                                                                                         ?column? 
----------
        1
(1 row)
```

File: antlr4/PostgreSQLParser.g4 [hunk 2]
```diff
@@ -1718,15 +1725,15 @@ non_reserved_keyword
...
+    |  FORCE | FORMAT | FORTRAN |  FORWARD |  FOUND |  FREE
     |  FUNCTION |  FUSION |  G_ |  GENERAL |  GENERATED
     |  GET |  GLOBAL |  GO |  GOTO | GREATEST | GRANTED
     |  GROUPING |  HANDLER |  HIERARCHY |  HOLD | HOST | HOUR
     |  IDENTITY |  IGNORE |  IMMEDIATE |  IMMUTABLE |  IMPLEMENTATION
     |  IMPLICIT |  INCLUDING |  INCREMENT |  INDEX |  INDICATOR
     |  INHERITS |  INOUT |  INPUT |  INSENSITIVE |  INSERT
     |  INSTANCE |  INSTANTIABLE |  INSTEAD |  INT |  INTEGER
-    |  INTERSECTION |  INTERVAL |  INVOKER |  ISOLATION
+    |  INTERSECTION |  INTERVAL |  INVOKER | ISNULL | ISNOTNULL ISOLATION | K_
```
Comment (tshprecher) [inline antlr4/PostgreSQLParser.g4]: If we need `ISNOTNULL` (see comment above), I think you missed a `|` between that and `ISOLATION`. Is there some test that needs it to exist?

File: antlr4/PostgreSQLParser.g4 [hunk 3]
```diff
@@ -1393,6 +1401,7 @@ expr
     | expr op=IS (bool_expr | NULL)
     | expr op=(ISNULL | NOTNULL)
     | expr IS NOT? DISTINCT FROM expr
+    | expr IS NOT? NULL
```
Comment (tshprecher) [inline antlr4/PostgreSQLParser.g4]: I don't have great insight into the slowdown in the select test cases, but I don't see anything effecting those tests in this PR except for `expr`. This change to `expr` seems simple and safe, but can we compare the timing of the tests with this change and with it commented? I know it sounds silly to test, but I do wonder...

As a side note, after we've reached basic correctness in the tests, `expr` is one of those rules that needs a top to bottom review. It has things like `expr op=IS (bool_expr | NULL)` along with `expr IS NOT? NULL`, and they be consolidated along with other things.

Comment (tshprecher) [reply antlr4/PostgreSQLParser.g4]: I thought about this some more and it brings back an old memory when I initially implemented this. I suspect this rule is triggering excess recursion through the case where `NOT` is include and the case where `NOT` is not included, which may trigger more recursion down the tree.

File: antlr4/PostgreSQLLexer.g4 (hunk 1 above)
Comment (mebelousov) [reply antlr4/PostgreSQLLexer.g4]: Hmm. I has reverted ISNOTNULL. It was seems that with ISNOTNULL is better. 
Thank you!

File: antlr4/PostgreSQLParser.g4 (hunk 3 above)
Comment (mebelousov) [reply antlr4/PostgreSQLParser.g4]: I has simplify `EXISTS OPEN_PAREN select_stmt CLOSE_PAREN`. Now tests run for ~ 1 minute. Maybe I will reduce the time of testing in the near future.

**Pull request merged**